io-engine-tests/src/file_io.rs

```diff
@@ -93,3 +126,65 @@ pub async fn compute_file_checksum(
 
     Ok(hex::encode(hasher.compute().0))
 }
+
+pub async fn compare_files(
+    path_a: impl AsRef<Path>,
+    path_b: impl AsRef<Path>,
+) -> std::io::Result<()> {
+    use std::io::{Error, ErrorKind};
+
+    let name_a = path_a.as_ref();
+    let name_b = path_b.as_ref();
+
+    let mut fa = OpenOptions::new()
+        .write(false)
+        .read(true)
+        .create(false)
+        .truncate(false)
+        .open(&path_a)
+        .await?;
+
+    let mut fb = OpenOptions::new()
+        .write(false)
+        .read(true)
+        .create(false)
+        .truncate(false)
+        .open(&path_b)
+        .await?;
+
+    let mut buf_a = [0; 16384];
+    let mut buf_b = [0; 16384];
+
+    let mut pos = 0;
+
+    loop {
+        let na = fa.read(&mut buf_a).await?;
+        let nb = fb.read(&mut buf_b).await?;
+
+        if na != nb {
+            return Err(Error::new(
+                ErrorKind::Other,
+                format!("Size of file {name_a:?} != size of {name_b:?}"),
+            ));
+        }
+
+        if na == 0 {
+            break;
+        }
+
+        for i in 0 .. na {
+            if buf_a[i] != buf_b[i] {
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    format!(
+                        "Miscompare at {} ({} blk): {:?} {:#02x} != {:?} {:#02x}",
+                        pos, pos / 512, name_a, buf_a[i], name_b, buf_b[i],
```

Comment:
This is assuming 512-bytes block size always? fio has configurable block size though.

Reply:
I'll remove it, it's confusing indeed.